Dmf/Modules.Library/Dmf_String.c

```diff
@@ -216,8 +287,202 @@ String_WideStringToMultiString(
 
     return multiString;
 }
+
+#pragma code_seg("PAGE")
+static
+NTSTATUS
+String_NarrowStringCopyAsUnicode(
+    _Out_ UNICODE_STRING* UnicodeString,
+    _In_z_ CHAR* NarrowString
+    )
+/*++
+
+Routine Description:
+
+    Copy a Narrow string as a Unicode string.
+
+Arguments:
+
+    UnicodeString - Target Unicode string.
+    NarrowString - Source Narrow string.
+
+Return Value:
+
+    NTSTATUS
+
+--*/
+{
+    NTSTATUS ntStatus;
+
+    PAGED_CODE();
+
+    FuncEntry(DMF_TRACE);
+
+    ASSERT(UnicodeString != NULL);
+    ASSERT(NarrowString != NULL);
+
+    WCHAR* wideString;
+
+    // Create a converted string.
+    //
+    wideString = String_MultiStringToWideString(NarrowString);
+    if (NULL == wideString)
+    {
+        ntStatus = STATUS_UNSUCCESSFUL;
+        TraceEvents(TRACE_LEVEL_ERROR, DMF_TRACE, "String_MultiStringToWideString");
+        goto Exit;
+    }
+
+    // Get length of converted string.
+    //
+    size_t stringLength = wcslen(wideString);
+
+    // Check to make sure that destinations ANSI string's buffer is big enough for 
+    // the string and zero terminator.
+    //
+    if (stringLength + sizeof(WCHAR) > UnicodeString->MaximumLength)
+    {
+        ntStatus = STATUS_BUFFER_TOO_SMALL;
+        TraceEvents(TRACE_LEVEL_ERROR, DMF_TRACE, "UnicodeString's buffer is too small");
+        goto Exit;
+    }
+
+    // Copy the converted string to the destination buffer.
+    //
+    wcscpy_s(UnicodeString->Buffer,

```

Comment:
I believe we are supposed to check the return value (errno_t) here.

Reply:
I will look into this next update.